File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-2B.cpp
//Write programs that read a sequence of integer inputs and print
//b.The number of even and odd inputs.

#include<iostream>
#include<vector>

using namespace std;

int main()
{
    cout << "Enter numbers: ";
    vector<int> numbers;

    bool exit = false;

    while(!exit)
    {
        int input;
        cin >> input;
        numbers.push_back(input);

        if(cin.fail())
        {
            exit = true;
            cout << "Stopping input.";
            // removes the faulty input so it doesn't get added to the odd/even counting
            numbers.pop_back();
        }
    }

    int odd = 0;
    int even = 0;

    for(int i = 0; i < numbers.size(); i++)
    {
        if(numbers[i] % 2 == 0)
        {
            even++;
        }

        else
        {
            odd++;
        }
    }


    // sugar-coating, not mandatory but looks better!
    if(odd != 1)
    {
        cout << "There are " << odd << " odd numbers\n";
    }

    else
    {
        cout << "There is " << odd << " odd number\n";
    }

    if(even != 1)
    {
        cout << "There are " << even << " even numbers\n";
    }

    else
    {
        cout << "There is " << even << " even number\n";
    }

    return 0;
}

File: Python For Everyone, Horstmann/Chapter4/P4.2B.py
# Write programs that read a sequence of integer inputs and print
#    a.  The smallest and largest of the inputs.
#    b.  The number of even and odd inputs.
#    c.  Cumulative totals. For example, if the input is 1 7 2 9, the program should print
#    1 8 10 19.
#    d.  All adjacent duplicates. For example, if the input is 1 3 3 4 5 5 6 6 6 2, the
#    program should print 3 5 6.

numEven = 0
numOdd = 0

stop = False

while not stop:
    inputN = str(input("Enter a number:(Stop/stop to stop): "))

    if inputN == "Stop" or inputN == "stop":
        stop = True

    elif inputN.isdigit():
        inputN = int(inputN)
        if inputN % 2 == 0:
            numEven += 1
        else:
            numOdd += 1

print("Number of even numbers:", numEven)
print("Number of odd numbers:", numOdd)